tests/testthat/test-derive_var_trtemfl.R

```diff
@@ -34,6 +34,44 @@ expected <- tibble::tribble(
 
 adae <- select(expected, -starts_with("TRTEM"))
 
+expected2 <- tribble(
+  ~USUBJID, ~ASTDTM, ~AENDTM, ~AEITOXGR, ~AETOXGR, ~AEGRPID, ~TRTEMFL, ~TRTEM2FL,
+  # before treatment
+  "1", "2021-12-13T20:15", "2021-12-15T12:45", "1", "1", "1", NA, NA,
+  "1", "2021-12-14T20:15", "2021-12-14T22:00", "1", "3", "1", NA, NA,
+  # starting before treatment and ending during treatment
+  "1", "2021-12-30T20:15", "2022-01-14T01:23", "3", "3", "2", NA, NA,
+  "1", "2022-01-05T20:00", "2022-06-01T11:00", "3", "1", "2", NA, NA,
+  "1", "2022-01-10T20:15", "2022-01-11T01:23", "3", "2", "2", "Y", "Y",
+  "1", "2022-01-13T20:15", "2022-03-01T01:23", "3", "1", "2", "Y", "Y",
+  # starting during treatment
+  "1", "2022-01-01T12:00", "2022-01-02T23:25", "4", "4", "3", "Y", "Y",
+
+  # after treatment
+  "1", "2022-05-10T11:00", "2022-05-10T13:05", "2", "2", "4", "Y", "Y",
+  "1", "2022-05-10T12:00", "2022-05-10T13:05", "2", "2", "4", NA, NA,
+  "1", "2022-05-11T11:00", "2022-05-11T13:05", "2", "2", "4", NA, NA,
```

Comment:
These two records should be flagged for test 3 as they are newly occurred AEs between `TRTSDT` and `(TRTEDT + 10)`. Could we add the left part of the following diagram during derivation? The right part (worsening) looks good to me.

```mermaid 
flowchart TB

AEITOXGR["AEITOXGR"] -->|"missing AEITOXGR: \n AE not exist before TRTSDT"| arise1["New AE records \n within TEAE window"] --> |New|TEAE[Step 3: Flag records identified above as TEAE]

AEITOXGR -->|"non-missing AEITOXGR:  \n AE exist before TRTSDT"|Worsen1["Step 1: Identify worsened records \n (compare with the previous record) \nafter TRTSDT"]--> Worsen2["Step 2: Get the first worsen date \n after TRTSDT"]-->worsen3["All records after \n the 1st worsen date \n and within TEAE window"]--> |Worsen|TEAE

style TEAE fill:#b3ffb3,stroke:#4CAF50,stroke-width:2px
style Worsen1 fill:#b3ffb3,stroke:#4CAF50,stroke-width:2px
style Worsen2 fill:#b3ffb3,stroke:#4CAF50,stroke-width:2px
```

Reply:
> These two records should be flagged for test 3 as they are newly occurred AEs between `TRTSDT` and `(TRTEDT + 10)`. Could we add the left part of the following diagram during derivation? The right part (worsening) looks good to me.
> 

I agree that the records are new AEs but are also grouped under one group_var. Records under group_var are compared to the previous intensity(except the first AE record), and in this case, the condition is not getting worse, so it's not flagged as TEAE. 
I only see it getting flagged if the first of the AEs under group_var is TEAE and subsequent records will also be flagged.

Reply:
[TEAE definition](https://database.ich.org/sites/default/files/E9_Guideline.pdf): An event that emerges during treatment having been absent pre-treatment, or worsens relative to the pre-treatment state.

- Left part: "Emerges during treatment, having been absent pre-treatment" – all newly occurring AEs (whether grouped or not) within the TEAE window should be flagged; no comparison is needed in this case.

- Right part: For AEs that existed pre-treatment, a comparison is needed to identify the worsening records.

Reply:
Agree that row 53 won't be flagged as it emerges after TRTEDT+10. I've addressed the remaining comments - please double-check. Thanks!

Reply:
It looks good to me, thanks a tonne for helping with this.